Oadp3355 1 3 1 release notes (#74026)
openshift/openshift-docs: modules/oadp-release-notes-1-3-1.adoc

link:https://issues.redhat.com/browse/OADP-3044[OADP-3044]

.Custom labels are not removed from the namespace, where OADP is installed

Previously, the `openshift-adp-controller-manager` pod would reset the labels on the namespace, where OADP is installed. Consequently, this caused synchronization issues for applications requiring custom labels such as Argo CD, leading to improper functionality. With this update, this issue is fixed and custom labels are not removed from the namespace, where OADP is installed.

link:https://issues.redhat.com/browse/OADP-3189[OADP-3189]

.OADP `must-gather` image collects CRD's

Previously, the OADP `must-gather` image did not collect the Custom Resource Definitions (CRD's) shipped by OADP. Consequently, it was not possible to use the `omg` tool to extract data on the support shell. For example, the `omg get backup` command did not work because `must-gather` was unable to collect CRD's.
With this fix, `must-gather` now collects CRD's shipped by OADP and can use the `omg` tool to extract data.

link:https://issues.redhat.com/browse/OADP-3229[OADP-3229]

.Garbage collection has the required default frequency

> Previously, the `garbage-collection-frequency` field had a wrong description for the default frequency. With this update, the `gc-controller` correctly executes its reconciliation loop every hour by default. As a result, garbage collection now has the expected default frequency.

Reviewer: the issue was with the description only
not the actaul value

Author: thank you, @amastbau, for reviewing it. I have updated it in this [commit](https://github.com/openshift/openshift-docs/pull/74026/commits/d39f89b0d244383f5e71b781477e0e98c44910a7 ).

// Module included in the following assemblies:
//
// * backup_and_restore/oadp-release-notes-1-3.adoc

:_mod-docs-content-type: REFERENCE
[id="oadp-release-notes-1-3-1_{context}"]
= OADP 1.3.1 release notes

The {oadp-first} 1.3.1 release notes lists new features and resolved issues.

[id="new-features-1-3-1_{context}"]
== New features

.OADP 1.3 Data Mover is now fully supported

The OADP built-in Data Mover, introduced in OADP 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.

> .{ibm-cloud-name} S3 backup storage provider is now supported

Reviewer 2: ```suggestion
.{ibm-cloud-name} is now supported as an AWS S3 compatible backup storage provider.
```

Reviewer 2: S3 is generally attached to AWS. I believe it should just be "{ibm-cloud-name}"
Please confirm if the term "{ibm-cloud-name} S3 " is valid.

Reviewer 3: IBM do offer `IBM Cloud Object Storage S3`, see [HERE](https://www.ibm.com/docs/en/watson-orchestrate?topic=storage-cloud-object-s3)

Author: @anarnold97 @xenolinux , could you please confirm if the following doc-text is ok for this-

> .{ibm-cloud-name} S3 is now supported as an AWS compatible backup storage provider
> 
> Previously, the AWS compatible object storage provider, {ibm-cloud-name} S3, was unsupported as backup storage locations.
> With this update, {ibm-cloud-name} S3 is now supported as an AWS compatible backup storage provider.

Thanks in advance!

Reviewer 3: @PrasadJoshi12 && @amastbau  - please can we have a QE opinion on this
Thanks

Reviewer 2: @apurvabhide17 looks good. Sorry I am not sure about this terminology.  I would just confirm with QE or SME if it should be "IBM Cloud Object Storage S3"?
In preview it's appearing as "IBM Cloud® S3"

Reviewer 4: +1 for IBM cloud Object Storage

Author: Hello @anarnold97, @xenolinux, @PrasadJoshi12 ,  
To  add "IBM cloud Object Storage" and avoid the confusion related to "the AWS compatible object storage provider, {ibm-cloud-name} S3, was unsupported", sharing my final draft here-

```
.{ibm-cloud-name} Object Storage is now supported as a backup storage provider

{ibm-cloud-name} Object Storage is one of the AWS S3 compatible backup storage provider, which was unsupported previously.  
With this update, {ibm-cloud-name} Object Storage is now supported as an AWS S3 compatible backup storage provider.
```
Please let me know so that I can add it and the PR will be ready for the merge process. Thanks!

Reviewer 2: LGTM - docs wise

Reviewer 2: @apurvabhide17 I think one change is reverted from the very initial suggestion. 

The phrase "one of the" is always followed by a plural noun. So, it should be "...one of the AWS S3 compatible backup storage **providers**"

Author: yup, my bad, I've updated it. Thanks :)

> Previously, {ibm-cloud-name} was one of the AWS S3 compatible object storage provider, which was unsupported for backup storage locations.

Reviewer 2: 1. I think "as backup storage locations" is accurate than "for backup storage locations"
2. IBM Cloud has always been S3 compatible provider so "IBM Cloud was one of the AWS...." can be avoided.

> With this update, {ibm-cloud-name} S3 is now supported as an AWS S3 compatible backup storage provider.

Reviewer 2: ```suggestion
With this update, {ibm-cloud-name} is now supported as an AWS S3 compatible backup storage provider.
```

[id="resolved-issues-1-3-1_{context}"]
== Resolved issues

> .IBM Cloud S3 backup storage provider is now supported

Reviewer 2: ```suggestion
.IBM Cloud backup storage provider is now supported as an AWS S3 compatible backup storage provider.
```
Same comment as above
IBM Cloud S3 ---- IBM Cloud

Previously, IBM Cloud was one of the AWS S3 compatible object storage provider, which was unsupported for backup storage locations.
With this update, IBM Cloud S3 is now supported as an AWS S3 compatible backup storage provider.

link:https://issues.redhat.com/browse/OADP-3788[OADP-3788]

.OADP operator now correctly reports the missing region error

Previously, when a user specified `profile:default` without specifying the `region` in AWS Backup Storage Location (BSL) configuration, the OADP operator failed to report the `missing region` error on the Data Protection Application (DPA) custom resource (CR). This update corrects validation of DPA BSL specification for the AWS provider. As a result, the OADP operator reports the `missing region` error.

link:https://issues.redhat.com/browse/OADP-3044[OADP-3044]

.Custom labels are not removed from the namespace, where OADP is installed

> Previously, the `openshift-adp-controller-manager` pod would reset the labels on the namespace, where OADP is installed. Consequently, this caused synchronization issues for applications requiring custom labels such as Argo CD, leading to improper functionality. With this update, this issue is fixed and custom labels are not removed from the namespace, where OADP is installed.

Reviewer 3: When OADP is installed, the default project (namespace) is `openshift-adp`.

link:https://issues.redhat.com/browse/OADP-3189[OADP-3189]

.OADP `must-gather` image collects CRD's

Previously, the OADP `must-gather` image did not collect the Custom Resource Definitions (CRD's) shipped by OADP. Consequently, it was not possible to use the `omg` tool to extract data on the support shell. For example, the `omg get backup` command did not work because `must-gather` was unable to collect CRD's.
With this fix, `must-gather` now collects CRD's shipped by OADP and can use the `omg` tool to extract data.

link:https://issues.redhat.com/browse/OADP-3229[OADP-3229]

.Garbage collection has the correct description for the default frequency value

> Previously, the `garbage-collection-frequency` field had a wrong description for the default frequency value. With this update, `garbage-collection-frequency` has the correct value for the `gc-controller` reconciliation default frequency, which is one hour. 

Author: Since "the `garbage-collection-frequency` field" is mentioned initially, I'm avoiding using the article in the next sentence along with the word "field".

link:https://issues.redhat.com/browse/OADP-3486[OADP-3486]

.Added FIPS Mode flag to the OADP Operator listing

> By setting the `fips-compliant` flag to `true`, the FIPS mode flag is now added to the OADP Operator listing in OperatorHub. This feature was enabled in OADP 1.3.0 but did not show up in the Red Hat Container catalog as being FIPS enabled.

Author: the doc-text is reviewed by the SME and QE, got an ack for it. Keeping the same.

With this fix, the following scenarios are possible:

* Sometimes the backup does not fail without any panic errors.
* The backup fails with the `Timed out awaiting reconciliation of volumesnapshot` timeout error and without panic.

link:https://issues.redhat.com/browse/OADP-3069[OADP-3069]

For a complete list of all issues resolved in this release, see the list of link:https://issues.redhat.com/issues/?filter=12432794[OADP 1.3.1 resolved issues] in Jira.

[id="known-issues-1-3-1_{context}"]
== Known issues

.Backup method limitation for {sno-caps} clusters on {ibm-power-name} and {ibm-z-name} platforms

The following are backup method restrictions for {sno-caps} clusters deployed on {ibm-power-name} and {ibm-z-name} platforms:
 
Storage requirement:: Only NFS storage is currently compatible with {sno} clusters deployed on {ibm-power-name} and {ibm-z-name} platforms.
Backup method:: Only the File System Backup method such as `kopia` and `restic` are supported for backup and restore operations.

link:https://issues.redhat.com/browse/OADP-3787[OADP-3787]

.Cassandra application has `CrashLoopBackoff` status after restore
 

> After OADP restores, the cassandra application has the `CrashLoopBackoff` status. To work around this problem, delete the `StatefulSet` pods with any error or with the `CrashLoopBackoff` state after restoring OADP.  The `StatefulSet` controller recreates these pods and they should run normally. 

Reviewer: ```suggestion
After OADP restores, the cassandra application  pods  might  have the  `CrashLoopBackoff` status. To work around this problem, delete the `StatefulSet` pods with any error or with the `CrashLoopBackoff` state after restoring OADP.  The `StatefulSet` controller recreates these pods and they should run normally. 
```

Author: done, thank you!

link:https://issues.redhat.com/browse/OADP-3229[OADP-3229]

.Garbage collection has the correct description for the default frequency

> Previously, the `garbage-collection-frequency` field had a wrong description for the default frequency. With this update, the `gc-controller` correctly executes its reconciliation loop every hour by default. As a result, garbage collection now has the expected description for the default frequency.

Reviewer: Previously, the `garbage-collection-frequency` field had a wrong description for the default frequency value. With this update,  'garbage-collection-frequency'   has the correct value of the  `gc-controller` reconciliation  default   frequency, which is one hour.
